task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,265 @@
+            GIT_URL=$(echo ${FIND_UNICODE_CONTROL_GIT_URL} | awk -F'#' '{print $1}')
+            REV=$(echo ${FIND_UNICODE_CONTROL_GIT_URL} | awk -F'#' '{print $2}')
+
+            # Clone find-unicode-control repository
+            git clone --depth=1 "${GIT_URL}" cloned >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to clone the repository: ${GIT_URL}" >&2
+                return 1
+            else
+                echo "Message: Succeed to clone the repository: ${GIT_URL}" >&2
+            fi
+
+            if [[ -n "${REV}" ]]; then
+                # Get git url suffix
+                git_url_suffix=$(git -C ./cloned/ rev-parse HEAD)
+                if [[ "${git_url_suffix}" != "${REV}" ]]; then
+                    (cd cloned && git checkout --quiet "${REV}")
+                fi
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}"
+            else
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}#${git_url_suffix}"
+            fi
+
+            # Check usage of find-unicode-control to confirm the tool can work
+            ./cloned/find_unicode_control.py --help >&2
+            status=$?
+            if [ "${status}" -ne 0 ]; then
+                echo "Error: Failed to check usage of find-unicode-control" >&2
+                return 1
+            else
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            fi
+        }
+
+        # Find unicode control
+        unicontrol_scan()
+        {
+            LANG=en_US.utf8 ./cloned/find_unicode_control.py -p bidi -v -d -t "${SOURCE_CODE_DIR}/source" \
+                >raw_sast_unicode_check_out.txt \
+                2>raw_sast_unicode_check_out.log \
+                || FUC_EXIT_CODE=$?
+
+            if [[ -z "${FUC_EXIT_CODE}" || "${FUC_EXIT_CODE}" -eq 1 ]]; then
+                # No non-printable unicode characters found, or successfully found
+                echo "Message: Succeed to run find-unicode-control command" >&2
+            else
+                # Failed to run the command
+                echo "Error: failed to run find unicode control command" >&2
+                return 1
+            fi
+
+            # Translate the output format
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|'
+            status=$?
+            if [ "${status}" -ne 0 ] && [ "${status}" -ne 1 ]; then
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            else
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --event='error|warning'
```

Comment:
The argument `--event='error|warning'` seems redundant because the `sed` command produces only `warning:` messages.

Reply:
Updated